### ISSUE TITLE
Incorporate more advanced minifiers, like Closure

### ISSUE THREAD

**Issue by westonruter** [xwp/wp-dependency-minification#7]
BeHeard reported an [issue](http://wordpress.org/support/topic/error-try-again?replies=8) on the WP forums, whereby dependency minification was failing on one or more scripts. The specific error was (see screenshot at end of description):

> JSMin: Unterminated RegExp at byte 230500: /g,"%2D%2D")+ Try again

It looks like a bug in JSMin, cause by the fact that it is not use a real parser: https://code.google.com/p/minify/issues/detail?id=222

So we should by default use JSMinPlus, and then fallback to JSMin, and if that does not work, throw the error described in #18. Note that I had defailed to JSMin since JSMinPlus was also failing to minify some script I was using during testing. We could use Google's Closure Compiler, but then we have to break up the code into chunks since Google imposes a limit on the POST size.

**Comment by nash-ye** [comment]
I want to help on this, To confirm you mean the Google Closure Compiler?
http://code.google.com/p/closure-compiler/

**Comment by westonruter** [comment]
@nash-ye Yes, Google's Closure Compiler. Thanks! I updated the issue description with more info.

Please base your work off of the current `develop` branch and not on top of your existing branch which is in a pending pull request.
